File: creational/prototype-example.py
class Person:
    def __init__(self, name, age):
        self.__name = name
        self.__age = age

    def to_str(self):
        return f"name: {self.__name}, age:{self.__age}"

    def inc_age(self):
        self.__age += 1

    def clone(self):
        temp = Person(self.__name, self.__age)
        return temp


if __name__ == "__main__":
    p1 = Person('kamran', 55)
    p2 = p1.clone()
    print(p1.to_str())
    print(p2.to_str())

    p1.inc_age()
    print(p1.to_str())
    print(p2.to_str())

File: creational/prototype-example.js
class Person {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    toString() {
        console.log(`My name is ${this.name} and I am ${this.age} years old.`);
    }

    increaseAge() {
        this.age += 1;
    }

    clone() {
        const temporary = new Person(this.name, this.age);
        return temporary;
    }
}

// ------------------------------------------------------------------
const p1 = new Person('Tina', 26);
p2 = p1.clone();

p1.toString();
p2.toString();

p1.increaseAge();

p1.toString();
p2.toString();